psi4/src/psi4/libmints/molecule.h

```diff
@@ -277,6 +277,8 @@ class PSI_API Molecule {
     double xyz(int atom, int _xyz) const;
     /// Returns mass atom atom
     double mass(int atom) const;
+    /// Checks whether atom is within bounds of atom_ or full_atom_
+    void check_atom_(int atom, bool full) const;
```

Comment:
By my understanding, because `check_atom_` is just a helper function local to the `.cc` file, it doesn't need to be declared in the header, and the definition in the `.cc` can drop the `Molecule::` part.

A rebase will fix Azure now, but we can also simply merge after the 3rd review since it'll heal in master.

Reply:
I don't think that works, as the function needs to access `atoms_` and `full_atoms_` which are only defined if the function belongs to the `Molecule::` class.

At least, with L281 commented and https://github.com/psi4/psi4/blob/7a892760e628f2d54ba02d6d09da71acf4c42c8c/psi4/src/psi4/libmints/molecule.cc#L2224 changed to:
```
void check_atom_(int atom, bool full) const  {
```

the compilation crashes with `error: 'full_atoms_' was not declared in this scope`.

Reply:
Sounds about like the depth of my C++ experience. Try for `protected:`, then, like https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/molecule.h#L155